<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ExperimentList: S4 Class for handling lists of Experiments â€¢ ExperimentList</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ExperimentList: S4 Class for handling lists of Experiments">
<meta property="og:description" content="ExperimentList">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ExperimentList</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/ExperimentList.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/ExperimentList/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ExperimentList: S4 Class for handling lists of
Experiments</h1>
                        <h4 data-toc-skip class="author">Dharmesh D.
Bhuva</h4>
            
            <h4 data-toc-skip class="date">15 March 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/ExperimentList/blob/HEAD/vignettes/ExperimentList.Rmd" class="external-link"><code>vignettes/ExperimentList.Rmd</code></a></small>
      <div class="hidden name"><code>ExperimentList.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The ExperimentList package defines S4 classes to handle data
      from multiple experiments or studies by providing features of
      lists as well as those of a concatenated experiment. Individual
      experiments can be in the form of SummarizedExperiment, Ranged
      SummarizedExperiment, SingleCellExperiment, or SpatialExperiment
      objects. Annotations specific to each experiment are stored thus
      providing a unified interface to dealing with data from multiple
      studies. Specialised functions to access experiment data, and to
      apply functions across experiments are implemented. Existing
      functions implemented for each individual experiment (e.g.,
      <code><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">SingleCellExperiment::reducedDim()</a></code>) can be readily
      applied across the entire list of experiments.</p>
    </div>
    
<div class="section level2">
<h2 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h2>
<p>The advent of high throughput molecular measurement technologies has
resulted in the generation of vast amounts of data. The
<code>SummarizedExperiment</code> object and its derivates have assisted
in hosting data from these technologies. The
<code>SingleCellExperiment</code> and <code>SpatialExperiment</code>
objects are able to store even higher resolution single-cell and spatial
transcriptomics measurements from a single biological sample
respectively. Reduced costs has enabled the generation of these data
from multiple biological samples. Such data is not easily stored and
manupilated within a single object. Concatenation of objects can
partially resolve this data since it can be analysed in unison, however,
prevents for sample-wise analysis. Maintaining a list of objects would
allow object-wise analysis, but would hinder collective analysis.</p>
<p>The <code>ExperimentList</code> object is designed to fill in this
gap and allows storage and manipulation of multiple
<code>SpatialExperiment</code>, <code>SingleCellExperiment</code>,
<code>RangedSummarizedExperiment</code> or
<code>SummarizedExperiment</code> objects. It provides both list-like
and object-like functionality thus providing dynamic access to the data
as the needs arise. For example, when analysing multiple spatial
transcriptomic datasets, users may wish to compute reduced dimensions
(e.g., PCA) on each individual object before dataset integration and
compute a combined reduced dimension after. In such scenarios, having a
hybrid interface to the data is beneficial.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ExperimentList"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="anatomy-of-an-experimentlist">Anatomy of an <code>ExperimentList</code><a class="anchor" aria-label="anchor" href="#anatomy-of-an-experimentlist"></a>
</h2>
<p>The <code>ExperimentList</code> object borrows its structure from the
object it hosts. When dealing with a list of
<code>SpatialExperiment</code> objects, all slots from the object are
made available therefore the rows contain feature (e.g.,
gene/transcript) information, the columns represent individual
observations (e.g., visium spots), and multiple assays can be stored.
The individual objects (different shades in the schematic) are
concatenated to create a single <code>SpatialExperiment</code> object.
The mapping between the original sample and the concatenated columns are
maintained using and internal slot (<code>experimentIndex</code>).
Sample-specific annotation (NOT spot-specific) are held in a
<code>DataFrame</code> and are linked to the columns of the new
<code>ExperimentList</code> object.</p>
<p>The following schematic demonstrates the structure of the
ExperimentList object and lists some common accessors to interact with
the object.</p>
<div class="figure">
<img src="EL.jpg" alt=""><p class="caption">ExperimentList anatomy</p>
</div>
<p>Under the hood, a separate class is used to hold the list of objects
as described in the table below.</p>
<table class="table">
<thead><tr class="header">
<th>Object class</th>
<th>Matched <code>ExperimentList</code> class</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>SummarizedExperiment</code></td>
<td><code>SummarizedExperimentList</code></td>
</tr>
<tr class="even">
<td><code>RangedSummarizedExperiment</code></td>
<td><code>RangedSummarizedExperimentList</code></td>
</tr>
<tr class="odd">
<td><code>SingleCellExperiment</code></td>
<td><code>SingleCellExperimentList</code></td>
</tr>
<tr class="even">
<td><code>SpatialExperiment</code></td>
<td><code>SpatialExperimentList</code></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="constructing-an-experimentlist">Constructing an <code>ExperimentList</code><a class="anchor" aria-label="anchor" href="#constructing-an-experimentlist"></a>
</h2>
<p>The <code>TENxVisiumData</code> package contains 10X Visium data from
various human and mouse tissues. We will use the breast cancer IDC and
ILC datasets from this package to demonstrate the ExperimentList
container. These data contain measurements of 36601 transcripts measured
across 7785 and 4325 spots respectively.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/ExperimentList" class="external-link">ExperimentList</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/helenalc/TENxVisiumData" class="external-link">TENxVisiumData</a></span><span class="op">)</span>

<span class="co">#download data</span>
<span class="va">spe1</span> <span class="op">=</span> <span class="fu">TENxVisiumData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TENxVisiumData/man/TENxVisiumData.html" class="external-link">HumanBreastCancerIDC</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">spe2</span> <span class="op">=</span> <span class="fu">TENxVisiumData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TENxVisiumData/man/TENxVisiumData.html" class="external-link">HumanBreastCancerILC</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">#remove alt exps - these should be matched across exps (likewise for rownames)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">spe2</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">#create a list of objects</span>
<span class="va">spe_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  <span class="st">'HumanBreastCancerIDC'</span> <span class="op">=</span> <span class="va">spe1</span>,
  <span class="st">'HumanBreastCancerILC'</span> <span class="op">=</span> <span class="va">spe2</span>
<span class="op">)</span>
<span class="va">spe_list</span>
<span class="co">#&gt; $HumanBreastCancerIDC</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 36601 7785 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(7785): AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HumanBreastCancerILC</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 36601 4325 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(4325): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></code></pre></div>
<p>Given the individual objects, we first create a sample-specific
annotation table and subsequently use it to create an
<code>ExperimentList</code> object. With the data prepared, the object
can be constructed using the <code><a href="../reference/ExperimentList.html">ExperimentList()</a></code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#create some artificial experiment annotations</span>
<span class="va">experimentAnnotation</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  sex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Female'</span>, <span class="st">'Female'</span><span class="op">)</span>,
  age <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">65</span>, <span class="fl">68</span><span class="op">)</span>,
  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'HumanBreastCancerIDC'</span>, <span class="st">'HumanBreastCancerILC'</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">experimentAnnotation</span>
<span class="co">#&gt;                         sex age</span>
<span class="co">#&gt; HumanBreastCancerIDC Female  65</span>
<span class="co">#&gt; HumanBreastCancerILC Female  68</span>

<span class="co">#create ExperimentList objects</span>
<span class="va">el</span> <span class="op">=</span> <span class="fu"><a href="../reference/ExperimentList.html">ExperimentList</a></span><span class="op">(</span>experiments <span class="op">=</span> <span class="va">spe_list</span>, experimentData <span class="op">=</span> <span class="va">experimentAnnotation</span><span class="op">)</span>
<span class="va">el</span>
<span class="co">#&gt; ExperimentList with 2 SpatialExperiments</span>
<span class="co">#&gt; class: SpatialExperimentList </span>
<span class="co">#&gt; dim: 36601 12110 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(12110): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTCCATACAACT-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; experiments: 2</span>
<span class="co">#&gt; experimentNames (2): HumanBreastCancerIDC HumanBreastCancerILC</span>
<span class="co">#&gt; experimentData names (2): sex age</span></code></pre></div>
<p>The <code>ExperimentList</code> object can be created without
<em>experimentData</em> and with an unnamed list. When using a named
list, names of the list should be matched with rownames of
<em>experimentData</em>.</p>
</div>
<div class="section level2">
<h2 id="common-operations-on-experimentlist">Common operations on <code>ExperimentList</code><a class="anchor" aria-label="anchor" href="#common-operations-on-experimentlist"></a>
</h2>
<div class="section level3">
<h3 id="subsetting">Subsetting<a class="anchor" aria-label="anchor" href="#subsetting"></a>
</h3>
<p>Subsetting can be performed in a similar way to the parental
<code>SummarizedExperiment</code> object using the <code>[</code>
function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#subset the first five features and first three samples</span>
<span class="va">el</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt; ExperimentList with 2 SpatialExperiments</span>
<span class="co">#&gt; class: SpatialExperimentList </span>
<span class="co">#&gt; dim: 5 3 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(5): ENSG00000243485 ENSG00000237613 ENSG00000186092</span>
<span class="co">#&gt;   ENSG00000238009 ENSG00000239945</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(3): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACAGAGCGACTCCT-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; experiments: 2</span>
<span class="co">#&gt; experimentNames (2): HumanBreastCancerIDC HumanBreastCancerILC</span>
<span class="co">#&gt; experimentData names (2): sex age</span></code></pre></div>
<p>In addition to this, entire experiments can be subsetted using the
<code>[</code> function as below.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#subset the first five features and all columns from the second experiment</span>
<span class="va">el</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, , exp <span class="op">=</span> <span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt; ExperimentList with 1 SpatialExperiments</span>
<span class="co">#&gt; class: SpatialExperimentList </span>
<span class="co">#&gt; dim: 5 4325 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(5): ENSG00000243485 ENSG00000237613 ENSG00000186092</span>
<span class="co">#&gt;   ENSG00000238009 ENSG00000239945</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(4325): HumanBreastCancerILC.AAACAACGAATAGTTC-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTCCATACAACT-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; experiments: 1</span>
<span class="co">#&gt; experimentNames (1): HumanBreastCancerILC</span>
<span class="co">#&gt; experimentData names (2): sex age</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="getters-and-setters">Getters and setters<a class="anchor" aria-label="anchor" href="#getters-and-setters"></a>
</h3>
<p>The functions below can be used to access and set data in the
object.</p>
<ul>
<li>
<code><a href="../reference/ExperimentList-methods.html">nexp()</a></code>, <code><a href="../reference/ExperimentList-methods.html">experimentNames()</a></code>
</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#number of experiments</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">nexp</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#names of experiments</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">experimentNames</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt; [1] "HumanBreastCancerIDC" "HumanBreastCancerILC"</span></code></pre></div>
<ul>
<li><code><a href="../reference/ExperimentList-methods.html">experiments()</a></code></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get a list of individual experiments</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">experiments</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt; $HumanBreastCancerIDC</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 36601 7785 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(7785): AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HumanBreastCancerILC</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 36601 4325 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(4325): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></code></pre></div>
<ul>
<li>
<code><a href="../reference/ExperimentList-methods.html">experimentData()</a></code>, <code><a href="../reference/ExperimentList-methods.html">colData()</a></code>
</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get experiment annotations</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">experimentData</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 2 rows and 2 columns</span>
<span class="co">#&gt;                              sex       age</span>
<span class="co">#&gt;                      &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; HumanBreastCancerIDC      Female        65</span>
<span class="co">#&gt; HumanBreastCancerILC      Female        68</span>
<span class="co">#get column annotations</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/ExperimentList-methods.html">colData</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 1 column</span>
<span class="co">#&gt;                                                     sample_id</span>
<span class="co">#&gt;                                                   &lt;character&gt;</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAAGTATCTCCCA-1 HumanBreastCancerIDC1</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACACCAATAACTGC-1 HumanBreastCancerIDC1</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGAGCGACTCCT-1 HumanBreastCancerIDC1</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGGGTCTATATT-1 HumanBreastCancerIDC1</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGTGTTCCTGGG-1 HumanBreastCancerIDC1</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACATTTCCCGGATT-1 HumanBreastCancerIDC1</span>
<span class="co">#get column annotations merged with experiment annotations</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/ExperimentList-methods.html">colData</a></span><span class="op">(</span><span class="va">el</span>, experimentData <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 3 columns</span>
<span class="co">#&gt;                                                     sample_id         sex</span>
<span class="co">#&gt;                                                   &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAAGTATCTCCCA-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACACCAATAACTGC-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGAGCGACTCCT-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGGGTCTATATT-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGTGTTCCTGGG-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACATTTCCCGGATT-1 HumanBreastCancerIDC1      Female</span>
<span class="co">#&gt;                                               age</span>
<span class="co">#&gt;                                         &lt;numeric&gt;</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAAGTATCTCCCA-1        65</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACACCAATAACTGC-1        65</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGAGCGACTCCT-1        65</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGGGTCTATATT-1        65</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACAGTGTTCCTGGG-1        65</span>
<span class="co">#&gt; HumanBreastCancerIDC.AAACATTTCCCGGATT-1        65</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="apply">Apply<a class="anchor" aria-label="anchor" href="#apply"></a>
</h3>
<p>The <code><a href="../reference/ExperimentList-methods.html">elapply()</a></code> function can be used to apply functions to
individual objects.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#apply function</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">elapply</a></span><span class="op">(</span><span class="va">el</span>, <span class="va">dim</span><span class="op">)</span>
<span class="co">#&gt; $HumanBreastCancerIDC</span>
<span class="co">#&gt; [1] 36601  7785</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HumanBreastCancerILC</span>
<span class="co">#&gt; [1] 36601  4325</span></code></pre></div>
<p>If the return type is the same as the type of the individual
experiment objects, they will be combined into
<code>ExperimentList</code> object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get the first 100 spots</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">elapply</a></span><span class="op">(</span><span class="va">el</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; ExperimentList with 2 SpatialExperiments</span>
<span class="co">#&gt; class: SpatialExperimentList </span>
<span class="co">#&gt; dim: 36601 200 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(200): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.AACCCGAGCAGAATCG-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.AACCCTACTGTCAATA-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#&gt; experiments: 2</span>
<span class="co">#&gt; experimentNames (2): HumanBreastCancerIDC HumanBreastCancerILC</span>
<span class="co">#&gt; experimentData names (2): sex age</span></code></pre></div>
<p>This function allows the <code>ExperimentList</code> object to be
treated as a list. Since the parent class of the object is
<code>SpatialExperiment</code>, all functions that work for it
automatically work for the <code>ExperimentList</code> object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#extract image data for each object separately</span>
<span class="fu"><a href="../reference/ExperimentList-methods.html">elapply</a></span><span class="op">(</span><span class="va">el</span>, <span class="va">imgData</span><span class="op">)</span>
<span class="co">#&gt; $HumanBreastCancerIDC</span>
<span class="co">#&gt; DataFrame with 2 rows and 4 columns</span>
<span class="co">#&gt;               sample_id    image_id   data scaleFactor</span>
<span class="co">#&gt;             &lt;character&gt; &lt;character&gt; &lt;list&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; 1 HumanBreastCancerIDC1      lowres   ####   0.0247525</span>
<span class="co">#&gt; 2 HumanBreastCancerIDC2      lowres   ####   0.0247525</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HumanBreastCancerILC</span>
<span class="co">#&gt; DataFrame with 1 row and 4 columns</span>
<span class="co">#&gt;                sample_id    image_id   data scaleFactor</span>
<span class="co">#&gt;              &lt;character&gt; &lt;character&gt; &lt;list&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; 1 HumanBreastCancerILC..      lowres   ####   0.0247525</span>
<span class="co">#extract image data collectively</span>
<span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">imgData</a></span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 3 rows and 4 columns</span>
<span class="co">#&gt;                sample_id    image_id   data scaleFactor</span>
<span class="co">#&gt;              &lt;character&gt; &lt;character&gt; &lt;list&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; 1  HumanBreastCancerIDC1      lowres   ####   0.0247525</span>
<span class="co">#&gt; 2  HumanBreastCancerIDC2      lowres   ####   0.0247525</span>
<span class="co">#&gt; 3 HumanBreastCancerILC..      lowres   ####   0.0247525</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="coercion">Coercion<a class="anchor" aria-label="anchor" href="#coercion"></a>
</h3>
<p><code>ExperimentList</code> objects can be coerced to their parental
classes or <code>ExperimentList</code> versions of their parental
classes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#convert to SpatialExperiment</span>
<span class="fu">as</span><span class="op">(</span><span class="va">el</span>, <span class="st">'SpatialExperiment'</span><span class="op">)</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 36601 12110 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span>
<span class="co">#&gt;   ENSG00000277196</span>
<span class="co">#&gt; rowData names(1): symbol</span>
<span class="co">#&gt; colnames(12110): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTCCATACAACT-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>
<span class="co">#convert to SummarizedExperiment</span>
<span class="fu">as</span><span class="op">(</span><span class="va">el</span>, <span class="st">'SummarizedExperiment'</span><span class="op">)</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 36601 12110 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames: NULL</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(12110): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTCCATACAACT-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#convert to SummarizedExperimentList</span>
<span class="fu">as</span><span class="op">(</span><span class="va">el</span>, <span class="st">'SummarizedExperimentList'</span><span class="op">)</span>
<span class="co">#&gt; ExperimentList with 2 SummarizedExperiments</span>
<span class="co">#&gt; class: SummarizedExperimentList </span>
<span class="co">#&gt; dim: 36601 12110 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames: NULL</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(12110): HumanBreastCancerIDC.AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt;   HumanBreastCancerIDC.AAACACCAATAACTGC-1 ...</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTCCATACAACT-1</span>
<span class="co">#&gt;   HumanBreastCancerILC.TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(1): sample_id</span>
<span class="co">#&gt; experiments: 2</span>
<span class="co">#&gt; experimentNames (2): HumanBreastCancerIDC HumanBreastCancerILC</span>
<span class="co">#&gt; experimentData names (2): sex age</span></code></pre></div>
<p>The full coercion hierarchy can be explored using the
<code>is()</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">is</span><span class="op">(</span><span class="va">el</span><span class="op">)</span>
<span class="co">#&gt;  [1] "SpatialExperimentList"      "SpatialExperiment"         </span>
<span class="co">#&gt;  [3] "ExperimentList"             "SingleCellExperiment"      </span>
<span class="co">#&gt;  [5] "RangedSummarizedExperiment" "SummarizedExperiment"      </span>
<span class="co">#&gt;  [7] "RectangularData"            "Vector"                    </span>
<span class="co">#&gt;  [9] "Annotated"                  "vector_OR_Vector"</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2022-03-10 r81874)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] TENxVisiumData_1.3.0        ExperimentHub_2.3.5        </span>
<span class="co">#&gt;  [3] AnnotationHub_3.3.9         BiocFileCache_2.3.4        </span>
<span class="co">#&gt;  [5] dbplyr_2.1.1                ExperimentList_0.99.0      </span>
<span class="co">#&gt;  [7] SpatialExperiment_1.5.4     SingleCellExperiment_1.17.2</span>
<span class="co">#&gt;  [9] SummarizedExperiment_1.25.3 Biobase_2.55.0             </span>
<span class="co">#&gt; [11] GenomicRanges_1.47.6        GenomeInfoDb_1.31.4        </span>
<span class="co">#&gt; [13] IRanges_2.29.1              MatrixGenerics_1.7.0       </span>
<span class="co">#&gt; [15] matrixStats_0.61.0          S4Vectors_0.33.10          </span>
<span class="co">#&gt; [17] BiocGenerics_0.41.2        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] rjson_0.2.21                  ellipsis_0.3.2               </span>
<span class="co">#&gt;  [3] rprojroot_2.0.2               scuttle_1.5.0                </span>
<span class="co">#&gt;  [5] XVector_0.35.0                fs_1.5.2                     </span>
<span class="co">#&gt;  [7] bit64_4.0.5                   interactiveDisplayBase_1.33.0</span>
<span class="co">#&gt;  [9] AnnotationDbi_1.57.1          fansi_1.0.2                  </span>
<span class="co">#&gt; [11] R.methodsS3_1.8.1             sparseMatrixStats_1.7.0      </span>
<span class="co">#&gt; [13] cachem_1.0.6                  knitr_1.37                   </span>
<span class="co">#&gt; [15] jsonlite_1.8.0                png_0.1-7                    </span>
<span class="co">#&gt; [17] R.oo_1.24.0                   shiny_1.7.1                  </span>
<span class="co">#&gt; [19] HDF5Array_1.23.2              BiocManager_1.30.16          </span>
<span class="co">#&gt; [21] compiler_4.2.0                httr_1.4.2                   </span>
<span class="co">#&gt; [23] dqrng_0.3.0                   assertthat_0.2.1             </span>
<span class="co">#&gt; [25] Matrix_1.4-0                  fastmap_1.1.0                </span>
<span class="co">#&gt; [27] limma_3.51.5                  cli_3.2.0                    </span>
<span class="co">#&gt; [29] later_1.3.0                   htmltools_0.5.2              </span>
<span class="co">#&gt; [31] tools_4.2.0                   glue_1.6.2                   </span>
<span class="co">#&gt; [33] GenomeInfoDbData_1.2.7        dplyr_1.0.8                  </span>
<span class="co">#&gt; [35] rappdirs_0.3.3                Rcpp_1.0.8.2                 </span>
<span class="co">#&gt; [37] jquerylib_0.1.4               pkgdown_2.0.2.9000           </span>
<span class="co">#&gt; [39] vctrs_0.3.8                   Biostrings_2.63.1            </span>
<span class="co">#&gt; [41] rhdf5filters_1.7.0            DelayedMatrixStats_1.17.0    </span>
<span class="co">#&gt; [43] xfun_0.30                     stringr_1.4.0                </span>
<span class="co">#&gt; [45] beachmat_2.11.0               mime_0.12                    </span>
<span class="co">#&gt; [47] lifecycle_1.0.1               edgeR_3.37.0                 </span>
<span class="co">#&gt; [49] zlibbioc_1.41.0               BiocStyle_2.23.1             </span>
<span class="co">#&gt; [51] ragg_1.2.2                    promises_1.2.0.1             </span>
<span class="co">#&gt; [53] parallel_4.2.0                rhdf5_2.39.6                 </span>
<span class="co">#&gt; [55] yaml_2.3.5                    curl_4.3.2                   </span>
<span class="co">#&gt; [57] memoise_2.0.1                 sass_0.4.0                   </span>
<span class="co">#&gt; [59] stringi_1.7.6                 RSQLite_2.2.10               </span>
<span class="co">#&gt; [61] BiocVersion_3.15.0            desc_1.4.1                   </span>
<span class="co">#&gt; [63] filelock_1.0.2                BiocParallel_1.29.17         </span>
<span class="co">#&gt; [65] rlang_1.0.2                   pkgconfig_2.0.3              </span>
<span class="co">#&gt; [67] systemfonts_1.0.4             bitops_1.0-7                 </span>
<span class="co">#&gt; [69] evaluate_0.15                 lattice_0.20-45              </span>
<span class="co">#&gt; [71] purrr_0.3.4                   Rhdf5lib_1.17.3              </span>
<span class="co">#&gt; [73] bit_4.0.4                     tidyselect_1.1.2             </span>
<span class="co">#&gt; [75] magrittr_2.0.2                R6_2.5.1                     </span>
<span class="co">#&gt; [77] magick_2.7.3                  generics_0.1.2               </span>
<span class="co">#&gt; [79] DelayedArray_0.21.2           DBI_1.1.2                    </span>
<span class="co">#&gt; [81] withr_2.5.0                   pillar_1.7.0                 </span>
<span class="co">#&gt; [83] prettydoc_0.4.1               KEGGREST_1.35.0              </span>
<span class="co">#&gt; [85] RCurl_1.98-1.6                tibble_3.1.6                 </span>
<span class="co">#&gt; [87] crayon_1.5.0                  DropletUtils_1.15.2          </span>
<span class="co">#&gt; [89] utf8_1.2.2                    rmarkdown_2.13               </span>
<span class="co">#&gt; [91] locfit_1.5-9.5                grid_4.2.0                   </span>
<span class="co">#&gt; [93] blob_1.2.2                    digest_0.6.29                </span>
<span class="co">#&gt; [95] xtable_1.8-4                  httpuv_1.6.5                 </span>
<span class="co">#&gt; [97] R.utils_2.11.0                textshaping_0.3.6            </span>
<span class="co">#&gt; [99] bslib_0.3.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Dharmesh D. Bhuva.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
